packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -842,21 +861,43 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   void _cursorTick(Timer timer) {
     _showCursor.value = !_showCursor.value;
+
+    if (_platformIsIOS) {
+      final double toValue = _showCursor.value ? 1.0 : 0.0;
+      _cursorController.animateTo(toValue, duration: _fadeDuration, curve: Curves.easeOut);
+    } else
+      _cursorController.value = _showCursor.value ? 1.0 : 0.0;
+
     if (_obscureShowCharTicksPending > 0) {
       setState(() { _obscureShowCharTicksPending--; });
     }
   }
 
+  void _cursorWaitForStart(Timer timer) {
+    _cursorTimer?.cancel();
+    _cursorTimer = Timer.periodic(_kCursorBlinkHalfPeriod, _cursorTick);
+  }
+
   void _startCursorTimer() {
     _showCursor.value = true;
-    _cursorTimer = Timer.periodic(_kCursorBlinkHalfPeriod, _cursorTick);
+    if (_platformIsIOS) {
+      _cursorColorTicker.start();
```

Comment:
you don't need both this an an animation controller right. You just want your _onCursorColorTick to listen to the animation controller instead.